examples/Raytracer_Benchmark/RaytraceBenchmark.cu

```diff
@@ -137,7 +137,42 @@ void RenderTiledImage(adept::BlockData<Ray_t> *rays, cuda::RaytracerData_t *rtda
   COPCORE_CUDA_CHECK(cudaGetLastError());
 }
 
-void initiliazeCudaWorld(cuda::RaytracerData_t *rtdata) {
+__global__ void getAllLogicalVolumes(vecgeom::VPlacedVolume *currentVolume, vecgeom::Vector<vecgeom::LogicalVolume *> *container)
```

Comment:
This looks quite inefficient as a way to extract the logical volumes in a vector. You don't need a (recursive) kernel doing it, you can try to use: CudaManager::LookupLogical(LogicalVolume const *const host_ptr) that returns the device pointer for each logical volume on the host. So if you need device pointers you can have just a loop on host on the vector of logical volumes stored by GeoManager (```GeoManager::GetAllLogicalVolumes()```) and get the device pointer using CudaManager.